indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -520,7 +521,10 @@ public boolean add(final Task task) throws EntryExistsException
     try {
       Preconditions.checkState(active, "Queue is not active!");
       Preconditions.checkNotNull(task, "task");
-      Preconditions.checkState(tasks.size() < config.getMaxSize(), "Too many tasks (max = %s)", config.getMaxSize());
+      if (tasks.size() >= config.getMaxSize()) {
+        throw new TooManyTasksException(StringUtils.format("Too many tasks (maxQueueSize = %d), " +
```

Comment:
Instead of TooManyTasksException, you can create a failure type similar to Forbidden. This failure type can be called TooManyRequests. The failure will have an error code set to 429.

In here, you can create DruidException.fromFailure(TooManyRequests).forPersona(Operator).message(<the message>)

Reply:
The "dropping" is misleading. We are not dropping anything, just rejecting the extra task. So lets just say that and please add to "retry later or increase the limit" in the error message.

Reply:
Once you get rid of TooManyException class, the PR will also have less changes IMO

Reply:
+1

Reply:
Added all review changes suggested. Tested and attached screenshot.
<img width="600" alt="Screenshot 2023-11-21 at 5 05 10 PM" src="https://github.com/apache/druid/assets/11345667/67de8379-d8dd-4f0c-8c93-eaf07bc106e6">